bson/src/main/org/bson/LazyBSONObject.java

```diff
@@ -169,9 +167,6 @@ Object readValue(final BsonBinaryReader reader) {
                 return reader.readString();
             case BINARY:
                 byte binarySubType = reader.peekBinarySubType();
-                if (BsonBinarySubType.isUuid(binarySubType) && reader.peekBinarySize() == 16) {
```

Comment:
This is a backwards-breaking behavioral change that is not possible to deprecate first.  I think it's preferable to leaving JAVA_LEGACY here, which could bite people unexpectedly.